### PR TITLE
Fix navbar margin issue on small screens

### LINKED ISSUE #6: Navigation bar is having margin problems when resizing from medium-size to small-size screen.
Medium-size:

Screenshot 2024-12-04 at 10 38 56

Small-size: 

Screenshot 2024-12-04 at 10 39 09

Note how, in the small size, the menu has no margin with the border of the screen.

There is also a difference in the position of the logo in the top left corner, also probably related to margin or padding. This difference causes the logo to jump when the screen is manually resized by the user.

### PR DESCRIPTION
**Pull request by theishu4**
Resolve the issue with the navbar margin on small screens, ensuring proper alignment and spacing for a better responsive design experience. This will involve adjusting the margin properties to ensure that the navbar displays correctly on devices with smaller screen widths. Fixes #6 

## Before:
Screenshot 2024-12-05 181834

## After:
Screenshot 2024-12-05 182850

## Update Video:

https://github.com/user-attachments/assets/c10ea151-551f-4f9a-a264-6819c4393628

### CODE REVIEW

File: components/header.tsx [hunk 1]
```diff
@@ -39,7 +39,7 @@ export default function Header() {
   const [isOpen, setIsOpen] = useState(false)
 
   const toggleMenu = () => {
-    setIsOpen(!isOpen)
```
Comment (ceilican) [inline components/header.tsx:42]: Is this change really necessary? It was more compact and more intuitive before the change.

Comment (ceilican) [reply components/header.tsx:42]: Also, this is out of scope for the issue that is being solved. It is better to submit separate PRs for separate issues.

**Review by ceilican — COMMENTED** [review]
Have you checked whether these changes fix the issue where the logo's position changes when the screen is resized too?

File: components/header.tsx (hunk 1 above)
Comment (theishu4) [reply components/header.tsx:42]: Yeah, that wasn't really related to the issue, but it was a small change, so I added it. I'll be more careful next time. Thanks for the feedback—another lesson learned in open source☺️!

## Why I had made change?

Comment (theishu4) [reply components/header.tsx]: Change Updated!

**Pull request merged**